src/app/plugins/skill/services/users.service.ts

```diff
@@ -0,0 +1,57 @@
+import { HttpClient } from "@angular/common/http"
+import { Injectable } from "@angular/core"
+import { map } from "rxjs/operators"
+import { Observable } from "rxjs"
+import * as _ from "lodash"
+
+const API_END_POINTS = {
+  PROFILE_REGISTRY_V1: '/apis/proxies/v8/user/v1/read/',
+  PROFILE_REGISTRY_V2: '/apis/proxies/v8/api/user/v2/read'
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class UsersService {
+  constructor(private http: HttpClient) { }
+
+  getUserById(userid: string): Observable<any> {
+    if (userid) {
+      return this.http.get<any>(API_END_POINTS.PROFILE_REGISTRY_V1 + userid)
+        .pipe(map((data: any) => {
+          return this.getFormatedRequest(data.result.response)
+        }))
+    }
+    return this.http.get<any>(API_END_POINTS.PROFILE_REGISTRY_V2).pipe(map(resp => resp.profiledetails))
+  }
+
+  getFormatedRequest(data: any) {
+    const userDetails = {
```

Comment:
here you don't need to define you object define like that 
const userDetails :any = {}